addon/services/documents.js

```diff
@@ -77,4 +107,44 @@ export default class DocumentsService extends Service {
       throw new Error(response.statusText, response.status);
     }
   }
+
+  /**
+   * Clears the document selection
+   */
+  @action clearDocumentSelection() {
+    this.selectedDocuments = [];
+    this.updateRoute();
+  }
+
+  /**
+   * Checks if the document is selected
+   *
+   * @param {Object} doc an EmberData representation of a Document
+   * @returns {Boolean} If the document is selected
+   */
+  @action documentIsSelected(doc) {
+    return !!this.selectedDocuments.find((d) => d.id === doc.id);
+  }
+
+  /**
+   * Selects the document
+   * @param {Object} doc an EmberData representation of a Document
+   */
+  @action selectDocument(doc) {
+    if (!this.selectedDocuments.includes(doc)) {
+      this.selectedDocuments = [...this.selectedDocuments, doc];
+      this.updateRoute();
+    }
+  }
+
+  /**
+   * Removes a document from the document selection
+   * @param {Object} doc an EmberData representation of a Document
+   */
+  @action deselectDocument(doc) {
+    this.selectedDocuments = this.selectedDocuments.filter(
+      (d) => d.id !== doc.id
+    );
+    this.updateRoute();
+  }
```

Comment:
After re-reading this: Did you consider using a `Set` instead of an array?

Reply:
As discussed, implementing shift selection (especially finding the first element) is clumsy with sets.